great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -1056,23 +890,6 @@ def test_connection(self) -> None:
                 f"failed. Ensure the table exists and the user has access to select data from the table: {query_error}"
             ) from query_error
 
-    @override
-    def test_partitioner_connection(self) -> None:
```

Comment:
Do we want to move this to the batch config (but probably not call it automatically)?

Reply:
this is only implemented for SQL partitioners, so to me it felt like a half baked feature. It would make more sense to me if we just had a `BatchConfig.test_connection()` method which checks the configuration, including the partitioner.

Reply:
Yeah, makes sense to me.